File: LinkedList.java
import static java.lang.System.out;
import java.lang.String;

public class LinkedList {
  public static void main(String[] args) {
    
    int[] data = new int[] {3,6,9,2,5,8,1};
    Linked_list result = new Linked_list();
    
    result.insertAll(data);
    result.printAll();
    result.insert(4);
    result.insert(7);
    result.delete(9);
    result.delete(11);
    result.insert(90);
    result.printAll();
    result.search(14);
    
  }
}

class Element {
  public int value;
  public Element ref; 
  Element(int val){
    value = val;
  }
}

class Linked_list {
  private Element head;
  private Element tail;
  public void insert(int x){
    if(head == null){
      head = new Element(x);
      tail = head;
    } else {
      Element elem = new Element(x);
      tail.ref = elem;
      tail = elem;
    }
  }
  public void insertAll(int[] x_list){
    for(int i:x_list)
      insert(i);
  }
  public void delete(int target){
    Element current = head;
    if (current.ref != null){
      Element front = search_helper(target);
      if (front == null){
        out.println(target+" : no such element");
        return;
      }
      Element middle = front.ref;
      if (middle.ref != null) 
        front.ref = middle.ref;
      middle.ref = null;
    } else if (current.value==target && current!=null){
      head = null;
      tail = null;
    }
  }
  public void search(int target){
    if (search_helper(target) != null){
      out.println("value exist");
    } else {
      out.println("not exist");
    }
  }
  public Element search_helper(int target){
    Element current = head;
    while(true){
      if (current.ref == null) return null;
      if ((current.ref).value == target) return current;
      current = current.ref;
    }
  }
  public void printAll(){
    Element x = head;
    String print_list = "";
    while(true){
      print_list += x.value+" ";
      if (x.ref == null)
        break;
      x = x.ref;
    }
    out.println(print_list);
  }
}

File: List/LinkedList.py
class Element():
    def __init__(self, val, ref=None):
        self.value = val
        self.ref = ref

class LinkedList():
    def __init__(self):
        self.head = None
        self.tail = None
    def insert(self,x):
        if self.head == None:
            self.head = Element(x)
            self.tail = self.head
        else:
            elem = Element(x)
            self.tail.ref = elem
            self.tail = elem
    def insertAll(self,x_list):
        for i in x_list:
            self.insert(i)
    def delete(self, target):
        current = self.head
        if current.ref != None:
            front = self.search_helper(target)
            if front==None:
                print(target,": No such element")
                return
            middle = front.ref
            if middle.ref:
                front.ref = middle.ref
            middle.ref = None
        elif current.value==target and current!=None:
            self.head = None
            self.tail = None
            
    def search(self, target):
        if self.search_helper(target)!=None:
            print("Value exist")
        else:
            print("not exist")
        
    def search_helper(self, target):
        current = self.head
        while (True) :
            if current.ref == None: # not found
                return None
            if (current.ref).value == target: # found
                return current 
            current = current.ref
            
    def printAll(self):
        x = self.head
        print_list = []
        while (True):
            print_list.append(x.value)
            if x.ref == None:
                break
            x = x.ref
        print(print_list)

data = [3,6,9,2,5,8,1]
result = LinkedList()
result.insertAll(data)
result.insert(4)
result.printAll()
result.delete(9)
result.insert(5)
result.insert(36)
result.delete(8)
result.printAll()
result.search(14)









